### ISSUE TITLE
Volume creation succeeds on a datastore which is in maintenance mode 

### ISSUE THREAD

**Issue by ashahi1** [vmware/docker-volume-vsphere#1651]
Steps:
```
1. Created a datastore cluster of two datastores - sharedVmfs-1 and sharedVmfs-0
2. Put sharedVmfs-1 in 'Maintenance Mode'
3. Then was able to create volume to the datastore  - sharedVmfs-1
4. Successfully mounted the volume to a container and was also able to write data to it.
```

This should not be allowed. Volume creation should not succeed on datastore which is in maintenance mode as user may have put it in maintenance mode to perform some service on it or user may even decide to drop the datastore completely which is in maintenance mode. 

Logs have been attached at the end.

-----------------------

Steps and their output are as follows:

1. Create a volume
```
root@sc-rdops-vm02-dhcp-52-237:~# docker volume create -d vsphere:latest --name TestVolMM@sharedVmfs-1
TestVolMM@sharedVmfs-1
root@sc-rdops-vm02-dhcp-52-237:~#
root@sc-rdops-vm02-dhcp-52-237:~#
root@sc-rdops-vm02-dhcp-52-237:~# docker volume ls
DRIVER              VOLUME NAME
vsphere:latest      TestVolMM@sharedVmfs-1
vsphere:latest      TestVolSS@sharedVmfs-0
vsphere:latest      VolSS1@sharedVmfs-0
root@sc-rdops-vm02-dhcp-52-237:~#

```
2. Docker volume inspect
```
root@sc-rdops-vm02-dhcp-52-237:~# docker volume inspect TestVolMM@sharedVmfs-1
[
    {
        "Driver": "vsphere:latest",
        "Labels": {},
        "Mountpoint": "/mnt/vmdk/TestVolMM@sharedVmfs-1",
        "Name": "TestVolMM@sharedVmfs-1",
        "Options": {},
        "Scope": "global",
        "Status": {
            "access": "read-write",
            "attach-as": "independent_persistent",
            "capacity": {
                "allocated": "13MB",
                "size": "100MB"
            },
            "clone-from": "None",
            "created": "Thu Jul 27 19:30:13 2017",
            "created by VM": "ubuntu-VM0.1",
            "datastore": "sharedVmfs-1",
            "diskformat": "thin",
            "fstype": "ext4",
            "status": "detached"
        }
    }
]

```
3. Mount the volume to a container and write some data
```
root@sc-rdops-vm02-dhcp-52-237:~# docker run -it --volume-driver=vsphere:latest -v TestVolMM@sharedVmfs-1:/vol1 --name ubMM ubuntu
root@0a16291ffec1:/#
root@0a16291ffec1:/# echo "Hello" > vol1/hellos2
root@0a16291ffec1:/# cat vol1/hellos2
Hello
root@0a16291ffec1:/#

```

Logs:
[docker-volume-vsphere.txt](https://github.com/vmware/docker-volume-vsphere/files/1181050/docker-volume-vsphere.txt)
[vmdk_ops.txt](https://github.com/vmware/docker-volume-vsphere/files/1181049/vmdk_ops.txt)

**Comment by lipingxue** [comment]
@ashahi1 Can we create  and attach VMDK to VM using ESXi UI when the datastore is in  maintenance mode?

**Comment by ashahi1** [comment]
@lipingxue I tried from VC but I am not able to create VM on a datastore which is in maintenance mode. Datastore maintenance mode is a VC feature.
Also, there is no event generated in hostd logs when datastore is put in maintenance mode. So it looks like ESX side is unaware of datastore maintenance mode.

Dockvols folder on datatore which is in maintenance mode, does show created vmdk.

**Comment by lipingxue** [comment]
@ashahi1 Since esx is not aware of datastore in maintenance mode, and still can create vmdk on the datastore, no fix can be done. We should document it as known issue.

**Comment by shuklanirdesh82** [comment]
@ashahi1 as discussed offline please include this into the same PR where you are adding information for inter-op features.

**Comment by ashahi1** [comment]
Fixed as part of PR #1689
